linux/keyman-config/keyman_config/gnome_keyboards_util.py

```diff
@@ -1,23 +1,52 @@
 #!/usr/bin/python3
 import logging
 import os
+import subprocess
 from gi.repository import Gio
 
 from gi.overrides.GLib import Variant
 
 
 class GnomeKeyboardsUtil():
     def __init__(self):
-        self.input_sources = Gio.Settings.new("org.gnome.desktop.input-sources")
+        if os.environ.get('SUDO_USER'):
+            self.input_sources = None
+            self.is_sudo = True
+            logging.debug('Running with sudo. Real user: %s', os.environ.get('SUDO_USER'))
+        else:
+            self.input_sources = Gio.Settings.new("org.gnome.desktop.input-sources")
+            self.is_sudo = False
+            logging.debug('Running as regular user')
 
     def read_input_sources(self):
-        sourcesVal = self.input_sources.get_value("sources")
-        sources = self._convert_variant_to_array(sourcesVal)
+        if self.is_sudo:
+            process = subprocess.run(
+                ['sudo', '-H', '-u', os.environ.get('SUDO_USER'),
+                 'DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/%s/bus' % os.environ.get('SUDO_UID'),
+                 'gsettings', 'get', 'org.gnome.desktop.input-sources', 'sources'],
+                capture_output=True)
```

Comment:
This pattern is repeated once in this file and also twice in ibus_util.py. Could we refactor these into a single function?

In fact, the entire wrapper around Gio.Settings could possibly be factored to a single helper that works in sudo and normal user mode?